File: 0344-reverse-string/0344-reverse-string.java
class Solution {
    void reverse_solve(char[] s, int low, int high) {
        if(low > high) {
            return;
        }
        char temp =  s[low];
        s[low] = s[high] ;
        s[high] = temp;

        reverse_solve(s, low+1, high -1);
    }
    public void reverseString(char[] s) {
        reverse_solve(s, 0, s.length-1);
    }
}

File: 0344-reverse-string/0344-reverse-string.cpp
class Solution {
public:

    void swap(char &a , char &b) {
        char temp;
        temp = a;
        a = b ;
        b = temp;
    }
    void reverseString(vector<char>& s) {
        int size = s.size();  
        int low = 0; 
        int high  = size - 1;
        while(low <= high ) {
            swap(s[low], s[high]);
            low++;
            high--;
        }
    }
};